plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/expression/ExpressionConverter.java

```diff
@@ -0,0 +1,260 @@
+            // (e.g., TIMESTAMP for 1-arg, TIMESTAMP WITH TIME ZONE for 2/3-arg versions)
+            // No need to force a correct returnType here; expression should handle it.
+            finalHudiExpression = new HudiTrinoFromUnixFunctionExpression(arguments, hudiReturnType);
+        }
+        else if (HudiTrinoYearFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            checkArgumentSize(1, functionName.getName(), arguments);
+            finalHudiExpression = new HudiTrinoYearFunctionExpression(arguments, hudiReturnType);
+        }
+        else if (HudiTrinoMonthFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            checkArgumentSize(1, functionName.getName(), arguments);
+            finalHudiExpression = new HudiTrinoMonthFunctionExpression(arguments, hudiReturnType);
+        }
+        else if (HudiTrinoHourFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            checkArgumentSize(1, functionName.getName(), arguments);
+            finalHudiExpression = new HudiTrinoHourFunctionExpression(arguments, hudiReturnType);
+        }
+        else if (HudiTrinoDateFormatFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            finalHudiExpression = new HudiTrinoDateFormatFunctionExpression(arguments, hudiReturnType);
+        }
+        else if (HudiTrinoSubstringFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            finalHudiExpression = new HudiTrinoSubstringFunctionExpression(arguments, hudiReturnType);
+        }
+        else {
+            // TODO: represent as a generic function call if not mapped?
+            log.warn("Unhandled Trino function: " + functionName);
+            finalHudiExpression = new HudiTrinoFunctionExpression(functionName.getName(), arguments, hudiReturnType);
+        }
+        tryAddAsCandidate(finalHudiExpression);
+        return finalHudiExpression;
+    }
+
+    private void tryAddAsCandidate(Expression expression)
+    {
+        if (expression instanceof Predicate predicate && isCandidateForExpressionIndex(predicate)) {
+            this.candidateCollector.add(predicate);
+        }
+    }
+
+    private static void checkArgumentSize(int argumentSize, String functionName, List<Expression> arguments)
+    {
+        if (arguments.size() != argumentSize) {
+            throw new TrinoException(INVALID_ARGUMENTS, "Trino " + functionName + " function expects 1 argument, but got " + argumentSize);
+        }
+    }
+
+    private static boolean isCandidateForExpressionIndex(Predicate predicate)
+    {
+        if (predicate instanceof Predicates.BinaryComparison binaryComparison) {
+            Expression left = binaryComparison.getLeft();
+            if (left instanceof HudiTrinoFunctionExpression fnExpression) {
+                return CANDIDATE_FUNCTIONS_FOR_EI.contains(fnExpression.getName());
```

Comment:
If the predicate is in the form of `day(ts) > 1 and day(ts) < 10`, it seems like it is not a candidate for the expression index.  Only `day(ts) > 1` is.  Is that the case, i.e., complex or nested predicate with expression wrapped might not be eligible for expression index pruning?